### ISSUE TITLE
Warning with an extension installed through .gitpod.yml and disappearing with a refresh of the browser

### ISSUE THREAD

**Issue by jgueytat** [gitpod-io/gitpod#3470]
### Bug description

An extension ready on VSCode and VSCode Remote has troubles on GitPod with Theia or Code IDE. 

### Steps to reproduce

The setup to reproduce is available here :  https://gray-wasp-3ie8v2ys.ws-eu03.gitpod.io/

### Expected behavior

The point was to get the exact same configuration between VSCode, VSCode Remote and GitPod.

### Example repository

https://github.com/jgueytat/HelloWorld-in-Ada

The original issue can be found on the extension dashboard : https://github.com/WebFreak001/code-debug/issues/260.
You will find a bunch of pictures.

**To sum up :**

- VSCode and VSCode Remote can hit the breakpoint (with the settings "debug.allowBreakpointsEverywhere": true).
- GDB can hit the breakpoint in command line.
- The two IDE code or Theia can not.

Is there a troubleshooting guide to check if an extension is healthy on Theia / GitPod ?

Thanks for the great work anyway !

**Comment by akosyakov** [comment]
Please make sure to user VS Code integration in Gitpod: https://www.gitpod.io/docs/feature-preview/#default-ide We are planning to make it default since April and deprecate Theia.

**Comment by akosyakov** [comment]
With VS Code in Gitpod I can debug it, although syntax highlighting is missing. It does not seem to find ada extension:
<img width="1904" alt="Screenshot 2021-03-17 at 09 19 01" src="https://user-images.githubusercontent.com/3082655/111436089-e0226680-8701-11eb-8187-f1182fcbfbd6.png">

**Comment by jgueytat** [comment]
I was able to reach a breakpoint right now with the same https://gray-wasp-3ie8v2ys.ws-eu03.gitpod.io/ I was using yesterday and which was not working. And I tried to relaunch a debug and then... no breakpoint again.
I can not find the proper reproduction steps yet.

**Comment by akosyakov** [comment]
I tried your workspace and debugging works for me each time 🤷‍♂️

**Comment by jgueytat** [comment]
Ok. Which user configuration files could interact? Could the browser have any interaction?

**Comment by jgueytat** [comment]
I just ran three or four time the debugger in a row. I am able to reach the breakpoint from time to time.

**Comment by jgueytat** [comment]
> Please make sure to user VS Code integration in Gitpod: https://www.gitpod.io/docs/feature-preview/#default-ide We are planning to make it default since April and deprecate Theia.

I will stick to code for some times now.

**Comment by akosyakov** [comment]
Could you open dev tools and see whether you have any relevant errors from the debug extension? You can also check output -> remote extension host channel. Are you sure that you don't have any error notifications or errors in the debug console when debugging falis?

**Comment by jgueytat** [comment]
![NativeDebug_NoLog](https://user-images.githubusercontent.com/1152718/111439619-d7339400-8705-11eb-84f4-6a60bc77dea0.png)
I just don't see any line related to this extension in the Log (Remote Extension) window.

I just did two launch in a row with the "showDevDebugOutput": true option in the debugger.

![Breakpoint_OK](https://user-images.githubusercontent.com/1152718/111440974-25956280-8707-11eb-8f69-d0110c6862bb.png)
![Breakpoint_KO](https://user-images.githubusercontent.com/1152718/111440958-229a7200-8707-11eb-8dbf-9d4d741ffcb1.png)

I need to see if I get the same random behavior with VSCode and VSCode Remote.

**Comment by jgueytat** [comment]
Yes I found a warning !
![Warning](https://user-images.githubusercontent.com/1152718/111441834-fe8b6080-8707-11eb-987f-97f9748fe8d8.png)

It's like there is an interaction with two debuggers.

**Comment by jgueytat** [comment]
Thanks a lot for your investigation @akosyakov ! I was able to reproduce the bug with VSCode locally. We will keep the investigation there : https://github.com/WebFreak001/code-debug/issues/260

**Comment by jgueytat** [comment]
I would like tough to understand the "Matches multiple schemas when only one must validate." warning.

**Comment by jgueytat** [comment]
I did turn off the sync settings. It's seems troublesome to me.

**No Warning with C / C++ :**
https://github.com/jgueytat/HelloWorld-in-C
https://magenta-muskox-95z77380.ws-eu03.gitpod.io

**Warning with Ada :**
https://github.com/jgueytat/HelloWorld-in-Ada
https://pink-chickadee-boeb0mlt.ws-eu03.gitpod.io/

Enabling the Ada extension does make appear the warning. This extension is not supposed to play with the debugger...
I don't get it much.

**Comment by jgueytat** [comment]
> I did turn off the sync settings. It's seems troublesome to me.
> 
> **No Warning with C / C++ :**
> https://github.com/jgueytat/HelloWorld-in-C
> https://magenta-muskox-95z77380.ws-eu03.gitpod.io
> 
> **Warning with Ada :**
> https://github.com/jgueytat/HelloWorld-in-Ada
> https://pink-chickadee-boeb0mlt.ws-eu03.gitpod.io/
> 
> **=> Enabling the Ada extension does make the warning appear. This extension is not supposed to play with the debugger... I don't get it much.**

**Erratum :** I get the warning in the C/C++ and Ada projects. Both are working like a charm with VSCode WSL and VSCode Remote Container.

**On GitPod I have those two troubles :**

- the warning on the launch.json
- I can not get any breakpoint (missing nearly 99% of the breakpoints, missing maybe 1% locally)

**Comment by jgueytat** [comment]
**Current understanding :**

- **I only get the warning on GitPod** : I guess the troubles is related to a conflict with a built-in extensions on GitPod and https://github.com/AdaCore/ada_language_server/issues/643
- **The missing breakpoint is not directly related to GitPod** and seems to be related to some latency : I guess it will endup with some advices to extension publisher concerning the troubles that are less visible locally. https://github.com/WebFreak001/code-debug/issues/260

**Comment by akosyakov** [comment]
>  I guess the troubles is related to a conflict with a built-in extensions on GitPod 

We try to run VS Code as close as possible to desktop settings, i.e. almost the same set of built-in extensions. We only add yaml to lint .gitpod.yml, gitpod extension to integrate with ports/tasks and the github pull request extension when it is needed.

**Comment by jgueytat** [comment]
Ok I found a way to reproduce the warning. The troubles comes from GitPod itself and not from the extensions.
On my two projects, C/C++ and Ada. **The first workspace I open won't have any warning.**
**The second I open will have the warning.** It does not depend on Ada or C/C++. The only common point is the Native Debug extension and even with uninstalling the Native Debug extension in the second workspace, **the warning remains** (but I can not launch the program anymore).

The troubles seems to be related to some sharing between user workspaces.

**Comment by jgueytat** [comment]
**I made locally some tests :**

- the two projects opened in a single workspace on VSCode WSL => No warning
- any of the two projects opened in a container => No warning

**Comment by jgueytat** [comment]
**Bug reproduction found :** https://github.com/jgueytat/HelloWarning
As soon as the extension Native Debug is installed through .gitpod.yml you get the warning. The warning disappear when you refresh your browser.

I did remove the sync extensions also during my tests.

**Comment by jgueytat** [comment]
@akosyakov this one can be considered as a bug I think.
There is also the question **to avoid or not** to have to refresh the browser.
